tests/Unit/LegacyConverters/DomainToLegacyConverterTest.php

```diff
@@ -5,159 +5,122 @@
 namespace WMDE\Fundraising\MembershipContext\Tests\Unit\LegacyConverters;
 
 use PHPUnit\Framework\TestCase;
-use WMDE\Euro\Euro;
 use WMDE\Fundraising\MembershipContext\DataAccess\DoctrineEntities\MembershipApplication as DoctrineApplication;
 use WMDE\Fundraising\MembershipContext\DataAccess\LegacyConverters\DomainToLegacyConverter;
 use WMDE\Fundraising\MembershipContext\Domain\Model\Incentive;
 use WMDE\Fundraising\MembershipContext\Tests\Data\ValidMembershipApplication;
-use WMDE\Fundraising\PaymentContext\Domain\Model\Iban;
-use WMDE\Fundraising\PaymentContext\Domain\Model\PayPalData;
+use WMDE\Fundraising\PaymentContext\Domain\Model\LegacyPaymentData;
+use WMDE\Fundraising\PaymentContext\Domain\Model\LegacyPaymentStatus;
 
 /**
  * @covers \WMDE\Fundraising\MembershipContext\DataAccess\LegacyConverters\DomainToLegacyConverter
  */
 class DomainToLegacyConverterTest extends TestCase {
 
 	public function testWhenPersistingApplicationWithModerationFlag_doctrineApplicationHasFlag(): void {
-		$this->markTestIncomplete( 'This will work again when we update the legacy converters' );
 		$doctrineApplication = new DoctrineApplication();
 		$application = ValidMembershipApplication::newDomainEntity();
 		$application->markForModeration();
 
 		$converter = new DomainToLegacyConverter();
-		$converter->convert( $doctrineApplication, $application );
+		$converter->convert(
+			$doctrineApplication,
+			$application,
+			new LegacyPaymentData( 1, 1, 'UEB', [], LegacyPaymentStatus::EXTERNAL_BOOKED->value )
```

Comment:
This combination of payment type and status does not make sense: `UEB` has `LegacyPaymentStatus::BANK_TRANSFER` and `BEZ` has `LegacyPaymentStatus::DIRECT_DEBIT`.